site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -4,44 +4,300 @@ title: Upgrade FusionAuth
+
+Whichever approach you take, you should always test the upgrade in a staging environment to identify potential issues. This is particularly important on more complex upgrades involving multiple version jumps or if you have made customizations to FusionAuth. We recommend that the staging environment be as close as possible to your production environment. This includes using the same data, or at least the same data volume, as production. This enables you to test both functionality changes and data update performance times before committing to the upgrade on production.
+
+=== Migrating Configuration Between Environments
+
+There are a number of options for managing the configuration of your FusionAuth instance over time and promoting changes from one environment to another. These include:
+
+- Script the changes.
+- Terraform.
+- Pulumi.
+- Manual configuration via the UI.
+
+You can read more in link:/docs/v1/tech/admin-guide/configuration-management[the Configuration Management guide].
+
+=== Themes and Customizations
+
+Review custom themes and how they will be impacted by any FusionAuth updates. If you have made any customizations, you may need to update them to work with the new version of FusionAuth. You can use the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] project to perform diffs on the themes to identify any changes that need to be made.
+
+include::docs/v1/tech/installation-guide/_theme-upgrade.adoc[]
+
+== Upgrade Options
+
+=== Easiest Route
+
+The easiest upgrade route is to simply stop all your services, upgrade everything, and then bring it all back online. 
+
+This is a good option for single-instance development or QA systems, or for systems that can tolerate longer periods of downtime.
+
+If minimizing downtime is a goal, then rolling upgrades or a blue-green deployment might be more suitable. 
+
+=== Rolling Upgrade
+
+You can perform a rolling upgrade if you have three FusionAuth nodes running behind a load balancer. The nodes are upgraded one at a time and the load balancer is configured to direct traffic to the upgraded nodes as they become available for use. This allows you to upgrade FusionAuth with minimal downtime. 
+
+Because individual nodes are taken offline during the rolling update, the capacity of your installation is reduced during the upgrade.
+
+You can choose to run in-place upgrades on each node or to replace each node with a new node running the latest version of FusionAuth. In-place upgrades are potentially faster, but if something goes wrong, you may need to roll back the upgrade. Replacing nodes may be slower, but if something goes wrong, you can simply revert to the previous version of FusionAuth by pointing the load balancer to the old nodes.
```

Comment:
```suggestion
You can choose to run in-place upgrades on each node or to replace each node with a new node running the latest version of FusionAuth. In-place upgrades are simpler and faster because you don't have to spin up a new server, but if something goes wrong, rolling back the upgrade is more complex. Replacing nodes may be slower, but if something goes wrong, you can revert to the previous version of FusionAuth by pointing the load balancer to the old nodes.
```